flow/aggregates.py

```diff
@@ -0,0 +1,509 @@
+        """
+        # Since we should always clear the aggregates before registering
+        # aggregates because we append the aggregates directly.
+        self._aggregates.clear()
+        self._aggregate_ids.clear()
+        aggregated_jobs = self._make_aggregates(project)
+        self._create_nested_aggregate_list(aggregated_jobs, project)
+
+    def _create_nested_aggregate_list(self, aggregated_jobs, project):
+        """signac-flow internally assumes every aggregate to be a tuple of jobs.
+
+        This method converts every aggregate in ``aggregated_jobs``, which may be of
+        any type, returned from an aggregator_function using the instance of
+        ``_MakeAggregates`` to an aggregate of jobs as tuple.
+        """
+        def _validate_and_filter_job(job):
+            "Validate whether a job is eligible to be a part of an aggregate or not."
+            if job is None:
+                return False
+            elif job in project:
+                return True
+            else:
+                raise LookupError(f'The signac job {job.get_id()} not found'
+                                  f'in {project}')
+
+        for aggregate in aggregated_jobs:
+            try:
+                filter_aggregate = tuple(filter(_validate_and_filter_job, aggregate))
+            except TypeError:  # aggregate is not iterable
+                ValueError("Invalid aggregator_function provided by the user.")
+            # Store aggregate in this instance
+            self._aggregates.append(filter_aggregate)
+            # Store aggregate by their ids in order to search through id
+            self._aggregate_ids[get_aggregate_id(filter_aggregate)] = filter_aggregate
+
+
+class _DefaultAggregateStore:
+    """This class holds the information of the project associated with
+    an operation function aggregated by the default aggregates i.e.
+    :py:class:`aggregator.groupsof(1)`.
+
+    When iterated through it's instance, it yields a tuple of a single job from
+    the Project.
+
+    :param project:
+        A signac project used to fetch jobs for creating aggregates.
+    :type project:
+        :py:class:`flow.FlowProject` or :py:class:`signac.contrib.project.Project`
+    """
+    def __init__(self, project):
+        self._project = project
+
+    def __iter__(self):
+        for job in self._project:
+            yield (job,)
+
+    def __getitem__(self, id):
+        "Return a tuple of a single job via job id."
+        try:
+            return (self._project.open_job(id=id),)
+        except KeyError:
+            raise LookupError(f"Did not find job with id {id}.")
+
+    def __contains__(self, aggregate):
+        """Return whether the job is present in the project associated with this
+        instance of :py:class:`_DefaultAggregateStore`.
+        """
+        # signac-flow internally assumes every aggregate to be a tuple.
+        # Hence this method will also get a tuple as an input.
+        assert len(aggregate) == 1
```

Comment:
We should not assert that the length of something passed to `__contains__` here is one. It may be convenient to check for inclusion of an aggregate against all `_AggregateStore` objects default or not. If the length is not 1 then we just return false.